src/snowflake/snowpark/functions.py

```diff
@@ -6722,6 +6722,42 @@ def array_unique_agg(col: ColumnOrName, _emit_ast: bool = True) -> Column:
     return _call_function("array_unique_agg", True, c, _emit_ast=_emit_ast)
 
 
+@publicapi
+def size(col: ColumnOrName, _emit_ast: bool = True) -> Column:
+    """Returns the size of the input ARRAY, OBJECT or MAP. Returns NULL if the
+    input column does not match any of these types.
+
+    Args:
+        col: A :class:`Column` object or column name that determines the values.
+
+    Example::
+        >>> df = session.create_dataframe([([1,2,3], {'a': 1, 'b': 2}, 3)], ['col1', 'col2', 'col3'])
+        >>> df.select(size(df.col1), size(df.col2), size(df.col3)).show()
+        ----------------------------------------------------------
+        |"SIZE(""COL1"")"  |"SIZE(""COL2"")"  |"SIZE(""COL3"")"  |
+        ----------------------------------------------------------
+        |3                 |2                 |NULL              |
+        ----------------------------------------------------------
+        <BLANKLINE>
+    """
+    c = _to_col_if_str(col, "size")
+    v = to_variant(c)
+    return (
```

Comment:
same comment as in https://github.com/snowflakedb/snowpark-python/pull/2682 wrt to JIRA, and also using `_emit_ast=False` for internal logic calls.